bb8/src/inner.rs

```diff
@@ -244,3 +271,39 @@ impl<M: ManageConnection> Reaper<M> {
         }
     }
 }
+
+struct SharedPoolInnerStatistics {
+    gets: AtomicU64,
+    gets_waited: AtomicU64,
+}
+
+impl SharedPoolInnerStatistics {
+    fn new() -> Self {
+        Self {
+            gets: AtomicU64::new(0),
+            gets_waited: AtomicU64::new(0),
+        }
+    }
+
+    fn record_get(&self, with_contention: bool) {
+        self.gets.fetch_add(1, Ordering::SeqCst);
+
+        if with_contention {
```

Comment:
It would be more efficient to do one `fetch_add()` per `record_get()`, such that we record into uncontended or contended gets (the other info can be derived).

Reply:
So you mean, just making one single `fetch_add` for one or the other statistic, and when we return info about both of them we return the sum of both for returning the total gets?

Reply:
I think the info we return might as well be the info we gather, and then the downstream caller can still aggregate those numbers to calculate this number.